sqld-libsql-bindings/src/wal_hook.rs

```diff
@@ -60,46 +60,50 @@ impl WalMethodsHook {
     pub const METHODS_NAME: &'static [u8] = b"wal_hook\0";
 
     pub fn wrap(
-        underlying_methods: *const libsql_wal_methods,
+        default_methods: *mut libsql_wal_methods,
+        maybe_bottomless_methods: *mut libsql_wal_methods,
```

Comment:
implementation starts leaking all over the place, is there not a cleaner way to do this? It seems like being able to compose hooks would make this a whole lot more elegant

Reply:
Yeah, we should consider making it properly composable, this is essentially a hacky workaround to get it working quickly. On the other hand, we actually don't really compose anything now, because it's only the replication writer that uses bottomless, and it's only the reader that uses custom xFrames methods :sweat_smile: so that's mostly why I decided that this mild hack is worth it for the first iteration

Reply:
yeah, my understanding is that we might need some API work on libsql to make wal_methods compose properly.
We can discuss that later.